source/simulator/core.cc

```diff
@@ -2099,14 +2101,18 @@ namespace aspect
         // Check whether to terminate the simulation. The first part of the
         // pair indicates whether to terminate the execution; the second
         // indicates whether to do one more checkpoint:
-        const std::pair<bool,bool> termination = termination_manager.execute();
+        const bool should_terminate = time_handler.should_simulation_terminate_now();
+
+        const bool write_checkpoint_due_to_termination
+          = should_terminate && time_handler.need_checkpoint_on_terminate();
+
+        const bool checkpoint_written = maybe_write_checkpoint(last_checkpoint_time,
+                                                               write_checkpoint_due_to_termination);
 
-        const bool checkpoint_written = maybe_write_checkpoint(last_checkpoint_time, termination);
         if (checkpoint_written)
           last_checkpoint_time = std::time(nullptr);
 
-        // see if we want to terminate
-        if (termination.first)
+        if (should_terminate)
           break;
       }
     while (true);
```

Comment:
Isnt this equivalent to
```
}
while (should_terminate == false);
```

Reply:
Yes, but then I need to move should_terminate outside the large while loop. :-(

Reply:
Oh, ok, leave it then.

Reply:
I've long wished that the C++ gods had made the trailing `while (...)` clause part of the enclosed scope :-(